plugins/call_button/call_button.tsx

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {CSSProperties, useState, useEffect, useRef} from 'react';
+import {injectIntl, IntlShape} from 'react-intl';
+import classNames from 'classnames';
+
+import MenuWrapper from 'components/widgets/menu/menu_wrapper';
+import Menu from 'components/widgets/menu/menu';
+import {Constants} from 'utils/constants';
+
+import {Channel, ChannelMembership} from 'mattermost-redux/types/channels';
+import {PluginComponent} from 'types/store/plugins';
+
+import './call_button.scss';
+
+type Props = {
+    currentChannel: Channel;
+    channelMember?: ChannelMembership;
+    intl: IntlShape;
+    locale: string;
+    pluginCallComponents: PluginComponent[];
+    sidebarOpen: boolean;
+}
+
+function CallButton(props: Props) {
+    if (props.pluginCallComponents.length === 0) {
+        return null;
+    }
+
+    const {formatMessage} = props.intl;
+
+    const [active, setActive] = useState(false);
+    const [clickEnabled, setClickEnabled] = useState(true);
+
+    const prevSidebarOpen = ((val: boolean) => {
+        const ref = useRef<boolean>();
+        useEffect(() => {
+            ref.current = val;
+        });
+        return ref.current;
+    })(props.sidebarOpen);
+
+    useEffect(() => {
+        if (prevSidebarOpen && !props.sidebarOpen) {
+            setClickEnabled(false);
+            setTimeout(() => {
+                setClickEnabled(true);
+            }, Constants.CHANNEL_HEADER_BUTTON_DISABLE_TIMEOUT);
+        }
+    }, [props.sidebarOpen]);
```

Comment:
I have to say that I'm not well-versed enough in the usage of `useRef` to deem this as correct, though it seems right to me. @michelengelen Would you be able to lend a hand or recommend a developer to review this piece?

Reply:
I am not 100% sure why this is written so complicated. If I understand correctly the purpose of this is to know if the sidebar has been opened before, right? It just seems overly complicated to me to use this IIFE and a `useEffect` call inside to achieve that.

I would suggest to try something like this:

```
    const prevSidebarOpen = useRef(props.sidebarOpen);

    useEffect(() => {
        if (prevSidebarOpen.current && !props.sidebarOpen) {
            setClickEnabled(false);
            setTimeout(() => {
                setClickEnabled(true);
            }, Constants.CHANNEL_HEADER_BUTTON_DISABLE_TIMEOUT);
        }
        prevSidebarOpen.current = props.sidebarOpen;
    }, [props.sidebarOpen]);
```

@mickmister using refs is a good way to store values over renders, without using state. refs do not trigger a re-render when changed, so you can freely manipulate them.

Reply:
> I am not 100% sure why this is written so complicated. 

@michelengelen 

That's almost certainly due to my attempt at doing something I didn't know anything about :p
My goal was to have a similar behaviour as to `componentDidUpdate()` where I get the `prevProps` which I want to compare. I googled a bit a around and got the impression I needed to write a custom hook. 

More than happy to simplify of course, thanks for the input there, will give it a try asap.

Reply:
Feel free to reach out if you need help with the UI 👍🏼

Reply:
@michelengelen Thanks, your suggestion works perfectly fine and it makes sense given we are using it in a single place and we already have a `useEffect` call in place.